### ISSUE TITLE
Clarify .connect() callback behavior

### ISSUE THREAD

**Issue by pfraze** [ssbc/pull-ws-server#12]
If you provide a callback as the 2nd parameter to `.connect()`, it does a [weird thing](https://github.com/ssbc/pull-ws-server/blob/master/client.js#L12) where the callback is used both in `onOpen` and `onClosed`, but only on one of them. What is it supposed to be? Is it supposed to be called after the connection is successfully created or fails to create, but not on `onClose` after a successful create?

**Comment by dominictarr** [comment]
it's an async callback, so it calls exactly once, either when it succeeds (onOpen) or fails (onClose)
though, looking at the code, i'm surprised it didn't `cb(null, stream)` on success.

**Comment by pfraze** [comment]
I see, the readme's example is what confused me:

```js
var stream = ws.connect('ws://localhost:9999', function (err) {
  if(err) throw err
  console.log('disconnected')
})
```

That usage had me thinking, the callback would be called when the connection closed. I think we should update it to:

```
var stream = ws.connect('ws://localhost:9999', function (err, stream) {
  if(err) throw err
  // stream is now active
})

**Comment by dominictarr** [comment]
yup, I agree.

**Comment by dominictarr** [comment]
fixed in 1.5.0
